### PR TITLE
Described our release procedure as Makefile

### PR DESCRIPTION
**Pull request by yoichitgy**
This PR addresses Issue #476.

- Made `Makefile` to describe our release procedure.
- Add a section for maintainers to `CONTRIBUTING.md` to tell about the `Makefile`.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -46,3 +46,9 @@ The installation of SwiftLint to your environment is optional. [Hound CI](https:
 Some Swift source files are generated from ERB files, e.g. `Resolver.swift` is generated from `Resolver.erb`, to easily maintain a set of functions that only differ the number of arguments. The generated source files are located under `GeneratedCode` group in the Xcode project. Do not modify the generated code directly, but instead modify the source ERB code. After modifying the ERB code, run `script/gencode` script to generate the Swift code.
 
 **Thank you for contributing!**
+
+## For Maintainers
+
+### Making a new release version
+
+Our release procedure is described as [Makefile](https://github.com/Swinject/Swinject/blob/master/Makefile). Run `make help` coomand for more info.
```
Comment (yoichitgy) [inline CONTRIBUTING.md:54]: The link to Makefile will be available after this PR is merged.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+# Make a release for Carthage.
+#
+VERSION_TAG=$(shell git describe --abbrev=0)
+.PHONY: carthage-release
+carthage-release:
+ifeq ($(shell which gh),)
+	$(error gh command not found. Install gh command and run `gh auth login` beforehand.)
+endif
+ifneq ($(VERSION_TAG),$(shell agvtool what-marketing-version -terse1))
+	$(error The versoin tag $(VERSION_TAG) does not match the version in Info.plist)
```
Comment (mpdifran) [inline Makefile:58]:
```suggestion
	$(error The version tag $(VERSION_TAG) does not match the version in Info.plist)
```

Comment (yoichitgy) [reply Makefile]: @mpdifran Thank you for finding and fixing the typo

**Comment by 1ucas** [comment]
I never released a Swinject version before. But it looks good to me!
Maybe we should take a look at an CD alternative, to ensure we don't forget to run any of the steps? 
Or even an hybrid solution, calling the Makefile during CD execution.
Example:
https://github.com/1ucas/Swinject/pull/3/files

**Review by mpdifran — APPROVED** [review]

**Comment by yoichitgy** [comment]
Thanks 1ucas! Yes, the release using CD is nice to maintain👍

**Comment by yoichitgy** [comment]
Thanks! I merged the PR! We can move to 1lucas's CD version later🚀

**Pull request merged**